adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/FeatureRDD.scala

```diff
@@ -136,6 +137,17 @@ object FeatureRDD {
   }
 
   /**
+   * Builds a FeatureRDD without SequenceDictionary information by running an
+   * aggregate to rebuild the SequenceDictionary.
+   *
+   * @param rdd The underlying Feature RDD to build from.
+   * @return Returns a new FeatureRDD.
+   */
+  private[rdd] def apply(rdd: RDD[Feature]): FeatureRDD = {
```

Comment:
No, I think this is wrong.  The intent was that callers should always use `inferSequenceDictionary`. This `MEMORY_ONLY` default is completely hidden from the caller and not documented anywhere.

Based on my comments above though, I think the `inferSequenceDictionary` method and others should be reworked so that we don't have default parameter values in any of the FooRDD constructors.  Let us provide multiple `apply` methods and have the caller be explicit about which one they are calling.